crates/cubecl-opt/src/version.rs

```diff
@@ -0,0 +1,194 @@
+use std::{
+    collections::{HashMap, HashSet},
+    mem::take,
+};
+
+use cubecl_core::ir::{Item, Operation, Variable};
+use petgraph::visit::EdgeRef;
+
+use crate::{ControlFlow, EdgeIndex, NodeIndex};
+
+use super::Optimizer;
+
+#[derive(Debug)]
+pub struct SsaState<'a> {
+    versions: HashMap<(u16, u8), u16>,
+    visited_blocks: &'a mut HashSet<NodeIndex>,
+    visited_edges: &'a mut HashSet<EdgeIndex>,
+    max_versions: &'a mut HashMap<(u16, u8), u16>,
+}
+
+#[derive(Debug, Clone)]
+pub struct PhiEntry {
+    pub block: NodeIndex,
+    pub value: Variable,
+}
+
+#[derive(Debug, Clone)]
+pub struct PhiInstruction {
```

Comment:
What is Phi refering to? If there is a link to some information about the concept, that would be great!